### PR TITLE
[SYCL][HIP] add device architecture for the GFX90C

### PR DESCRIPTION
**Pull request by jinz2014**
The PR may address https://github.com/intel/llvm/issues/12010.  Thanks for your review.

### CODE REVIEW

**Comment by al42and** [comment]
Would it make sense to add the rest of the "supported" (in a sense of being a valid LLVM target, not in a sense of AMD supporting ROCm/HIP on a specific hardware) architectures?

Looking at https://github.com/intel/llvm/blob/sycl/llvm/docs/AMDGPUUsage.rst#L381, it seems we are missing quite a few here: `gfx909`, `gfx940`, `gfx941`, `gfx942`, `gfx1033`, gfx1036`, `gfx1100`, `gfx1101`, `gfx1102`, `gfx1103`, `gfx1150`, `gfx1151`

File: sycl/include/sycl/ext/oneapi/experimental/device_architecture.hpp [hunk 1]
```diff
@@ -449,6 +453,8 @@ struct IsAOTForArchitectureClass {
         __SYCL_TARGET_AMD_GPU_GFX908__ == 1;
     arr[static_cast<int>(arch::amd_gpu_gfx90a)] =
         __SYCL_TARGET_AMD_GPU_GFX90A__ == 1;
+    arr[static_cast<int>(arch::amd_gpu_gfx90c)] =
+        __SYCL_TARGET_AMD_GPU_GFX90C__ == 1;
```
Comment (dm-vodopyanov) [inline sycl/include/sycl/ext/oneapi/experimental/device_architecture.hpp:457]: Please also add a support of 
https://github.com/intel/llvm/blob/6e2d9f7144f1f6b52f077a436945c0ad9546fa16/sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc#new-member-function-of-device-class
and
https://github.com/intel/llvm/blob/6e2d9f7144f1f6b52f077a436945c0ad9546fa16/sycl/doc/extensions/experimental/sycl_ext_oneapi_device_architecture.asciidoc#new-device-descriptor
for `arch::amd_gpu_gfx90c`

For that, please update `NVIDIA_AMD_ARCHES`:
https://github.com/intel/llvm/blob/974cc70f9160e7002f4dd3774ae0556ee34a4cb8/sycl/source/detail/device_info.hpp#L588

Comment (jinz2014) [reply sycl/include/sycl/ext/oneapi/experimental/device_architecture.hpp:457]: Thanks.

**Review by dm-vodopyanov — APPROVED** [review]
`sycl/include` and `sycl/source` LGTM

**Review by mdtoguchi — APPROVED** [review]
driver changes LGTM.  I have no comment on the other target additions - are they commonly used?

File: sycl/source/detail/device_info.hpp [hunk 2]
```diff
@@ -588,6 +588,7 @@ struct get_device_info_impl<range<Dimensions>,
   X("gfx906", oneapi_exp_arch::amd_gpu_gfx906)                                 \
   X("gfx908", oneapi_exp_arch::amd_gpu_gfx908)                                 \
   X("gfx90a", oneapi_exp_arch::amd_gpu_gfx90a)                                 \
+  X("gfx90c", oneapi_exp_arch::amd_gpu_gfx90c)                                 \
```
Comment (dm-vodopyanov) [inline sycl/source/detail/device_info.hpp:591]: @jinz2014 compilation failed, `amd_gpu_gfx90c` should be added also to `enum class architecture` in sycl/include/sycl/ext/oneapi/experimental/device_architecture.hpp

Comment (jinz2014) [reply sycl/source/detail/device_info.hpp:591]: Thanks.

**Comment by al42and** [comment]
> Not a product manager, I have no idea of the usage. I will try to add all targets listed by @al42and.

I'm not a product manager either :) I just think that adding supported architectures one-by-one is not the best approach.

**Comment by jinz2014** [comment]
@al42and  All targets are added #12028 though I don't know how to add them in the best way. Hopefully, users could evaluate SYCL on a variety of AMD Radeon Graphics in their laptops and desktops.

**Comment by jinz2014** [comment]
Close the PR as it is part of #12028. Thank you.

**Pull request closed without merging**